### ISSUE TITLE
[QUESTION] During compilation, the makefile cannot find recipes for any target.

### ISSUE THREAD

**Issue by Asmitab13** [geoschem/geos-chem#158]
I am trying to run geoschem on a standalone system and I have downloaded the merra2 4x5 grid data.I have configured the run directory after generating it using the changes made to the copyRunDirs.input file. Which brings me to two issues;

The directory called gcgrid cannot be found when I use wget to download it, so right now my gcgridroot and the data root are the same path. Is that a problem? 
Assuming that this is not a problem, I have set all the environment variables for the fortran compilers and downloaded netcdf libraries using spack. The error I keep running into when I run "**make -j4 build**" is as follows:
**Badly placed ()'s.
Badly placed ()'s.
make[1]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
make[1]: Entering directory '/home/user/Documents/GC/rundirs/merra2_4x5_standard'
Badly placed ()'s.
Badly placed ()'s.
rm -f *.log
rm: No match.
Makefile:429: recipe for target 'cleanup_logs' failed
make[1]: *** [cleanup_logs] Error 1
make[1]: Leaving directory '/home/user/Documents/GC/rundirs/merra2_4x5_standard'
Makefile:326: recipe for target 'build' failed
make: *** [build] Error 2**

I realize that the j4 is just because I am running it on a standalone system but I am suspecting that there is some other versioning error that is preventing the makefile from finding these recipes for the targets. 

NOTE:- Also, the hemco data is really massive so I haven't downloaded it yet, is that really necessary for these steps? As in, does it contain the recipes needed for compilation?

Any help would be appreciated.

Thanks,
Asmita

**Comment by msulprizio** [comment]
What version of GEOS-Chem are you using? Can you also provide information about your environment (e.g fortran compiler version, netCDF library version)? Can you first try doing `make realclean` then `make build` without the `-j4` option and let us know if either of those succeeds?

As for the HEMCO data, you do not need to download those at this point if you are just trying to compile. You will need to download the HEMCO emissions data, in addition to the GEOS-Chem met fields once you're ready to run your GEOS-Chem simulation. Please see [this wiki page](http://wiki.seas.harvard.edu/geos-chem/index.php/Downloading_GEOS-Chem_data_directories) for details.

**Comment by Asmitab13** [comment]
Hi Melissa,

I am using GEOS-Chem 12.6.2, my fortran compiler version is 7.4.0 and my netCDF library version (that I installed using Spack) is 4.7.3.
I tried doing make realclean but I still get the error:
make: *** No rule to make target 'build'.  Stop.

Thank you very much for you help.

Best,
Asmita.

**Comment by yantosca** [comment]
Hi Asmita,  Thanks for writing.

I wonder if some of the Makefile variables, which are referenced with $(), are empty.  If you are using a Spack-built software stack, then you can use "spack load", which will place the paths to these libraries in certain environment variables like $PATH and $LD_LIBRARY_PATH:

```
spack load gcc
spack load netcdf
spack load netcdf-fortran
```

You might also have to add

```bash
export FC=gfortran
export CC=gcc
export CXX=g++
```

in your .bashrc file so that those variables get defined.

**Comment by yantosca** [comment]
Also -- if you are using GEOS-Chem 12.6.2 you can use CMake to build GEOS-Chem.   You can install CMake with Spack (CMake 3.15 is recommended).

To use CMake, type the following:
```bash
cd merra2_4x5_standard    # or whatever your run directory is
mkdir gcbuild
cd gcbuild
cmake ../CodeDir          # Configure GEOS-Chem
make -j4 install          # Build GEOS-Chem
```
If all goes well that should build the executable.  Otherwise it will report errors.

Then type:
```bash 
cd ..
./geos > log     #  runs GEOS-Chem, you can also submit a script to a scheduler
```

That should compile & run GEOS-Chem.

CMake is included as a "silent" option in GEOS-Chem 12.6.0+ but will become standard in GEOS-Chem 13.0.0.  You can find out more about CMake on the GEOS-Chem wiki at: http://wiki.geos-chem.org/CMake_FAQ

The nice thing about CMake is that if your Unix environment is not set up properly, CMake will find it during the configuration stage.
